### PR TITLE
Refactor CLIFF to use a plugin system for Entity Extraction

### PR DESCRIPTION
**Pull request by owaaa**
This is a fairly large Pull Request so let me know if you have any questions. The root goal is I wanted to not use the Stanford NER which is GNU licensed. However, rather than just forking this and go off on my own I tried to make CLIFF pluggable so people can choose their own Extraction method. Unfortunately, this necessitated refactoring a lot of how CLIFF is built. I had to pull much of CLIFF into a commons library jar, and then modules are used for the NER plugins. The servlet is now separate in the webapps profile. Also, by doing a mvn command in the project root, everything should hopefully work as normal using the Stanford NER model.

Basically, this refactor makes the entity extraction for CLIFF pluggable. This allows
the use of non-GNU extraction methods such as the MITIE library. It also
should allow the use of multiple targeted extraction methods. For example
if plugin one can support types X and Y, and plugin two type Z, this
should be able to merge the two entity sets.

### CODE REVIEW

**Comment by owaaa** [comment]
Cool let me know what you think. I'm the process of looking at some other tweeks as I'm testing some things. The other thing I'm considering is making the Focus Strategy configurable if I come up with an alternative strategy. 

I've also Docker'ized things but I focused on using MITIE. Let me know if you are interested in that. I've tried to keep my changes here as close to the original as possible just added features.
